src/Factory/Api/Mastodon/Account.php

```diff
@@ -60,31 +59,39 @@ public function __construct(LoggerInterface $logger, BaseURL $baseURL, ProfileFi
 	 */
 	public function createFromContactId(int $contactId, $uid = 0): \Friendica\Object\Api\Mastodon\Account
 	{
-		$cdata = Contact::getPublicAndUserContactID($contactId, $uid);
-		if (!empty($cdata)) {
-			$publicContact = Contact::getById($cdata['public']);
-			$userContact   = Contact::getById($cdata['user']);
-		} else {
-			$publicContact = Contact::getById($contactId);
-			$userContact   = [];
+		$contact = Contact::getById($contactId, ['uri-id']);
+		if (empty($contact)) {
+			throw new HTTPException\NotFoundException('Contact ' . $contactId . ' not found');
 		}
+		return self::createFromUriId($contact['uri-id'], $uid);
+	}
 
-		if (empty($publicContact)) {
-			throw new HTTPException\NotFoundException('Contact ' . $contactId . ' not found');
+	/**
+	 * @param int $contactUriId
+	 * @param int $uid          Public contact (=0) or owner user id
```

Comment:
> Public contact (=0) or owner user id

I'm surprised by this hint (also present in `createFromContactId`). They refer to two very different tables and it looks like it's not used with the contact table at all.

Reply:
I guess it means that with "uid=0" you receive the public contact or the contact entry for the user.